MACHINE-LEARNING-SERVICE/src/models/yolo.model.js

```diff
@@ -30,32 +31,39 @@ class Yolo extends ObjectDetection {
 
   // Allows to load the model and decode the image in order to make a detection of the desired object.
   async predict() {
-    const imagesArray = await fs.readdir(this.pathFile);
-    const yolo = yolov5;
-    await yolo.load();
+    
+    try {
+      const imagesArray = await fs.readdir(this.pathFile);
+      const yolo = yolov5;
+      await yolo.load();
 
-    const imagesToPredictArray = await Promise.all(
-      imagesArray.map(async (fileName) => {
-        const img = await fs.readFile(`${this.pathFile}${fileName}`);
-        const imgDecoded = tfnode.node.decodeImage(img, 3);
-        const imgResized = tf.image.resizeBilinear(imgDecoded, [640, 640]);
-        const image = tf
-          .cast(imgResized, 'float32')
-          .div(tf.scalar(255))
-          .expandDims(0);
-        const result = await yolo.predict(image);
-        const predict = yolo.getDetections(result);
-        const data = { predict, fileName };
-        return data;
-      })
-    );
+      const imagesToPredictArray = await Promise.all(
+        imagesArray.map(async (fileName) => {
+          const img = await fs.readFile(`${this.pathFile}${fileName}`);
+          const imgDecoded = tfnode.node.decodeImage(img, 3);
+          const imgResized = tf.image.resizeBilinear(imgDecoded, [640, 640]);
+          const image = tf
+            .cast(imgResized, 'float32')
+            .div(tf.scalar(255))
+            .expandDims(0);
+          const result = await yolo.predict(image);
+          const predict = yolo.getDetections(result);
+          const data = { predict, fileName };
+          return data;
+        })
+      ).catch(error =>{
+        throw new MachineLearningException('Error building model YOLO.', 'ML-02');
+      });
 
-    const foundObjectsArray = FilerResults.filterFunction(
-      imagesToPredictArray,
-      this.objectRequired,
-      this.percentage
-    );
-    return foundObjectsArray;
+      const foundObjectsArray = FilerResults.filterFunction(
+        imagesToPredictArray,
+        this.objectRequired,
+        this.percentage
+      );
+      return foundObjectsArray;
+    } catch (error) {
+      throw error;
```

Comment:
Should throw a specif error mensage

Reply:
The error is thrown in the promise.catch and propagated in the try-catch to send the message from the controller